src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -228,7 +228,11 @@ public bool IsSetAsInput
 
             set
             {
-                isSetAsInput = value;
+                if (isSetAsInput != value)
+                {
+                    isSetAsInput = value;
+                    this.OnNodeModified();
```

Comment:
@mjkkirschner  @QilongTang Is there a better even that can be raised here?  I believe this event will force re-execution of the graph.  I tried `MarkNodeAsModified(false)` which should mark as modified but not force execution as the AST is unchanged but it doesn’t seem to mark the graph as modified.

Reply:
I think I need @aparajit-pratap 's input here, even though AST should be updated here, should the graph re-run? I don't see changing if a node is input or output would affect graph results, but maybe we should..

Reply:
Yeah I wasn't sure about how clients would monitor these events and if re-execution is required say for Refinery.